src/main/java/com/marklogic/kafka/connect/source/KeyGenerator.java

```diff
@@ -0,0 +1,15 @@
+package com.marklogic.kafka.connect.source;
+
+import java.util.Map;
+
+public interface KeyGenerator {
+    String getUniqueKey();
+
+    static KeyGenerator newKeyGenerator(Map<String, Object> parsedConfig, Long rowNumber) {
+        if(!parsedConfig.get(MarkLogicSourceConfig.KEY_STRATEGY).toString().equalsIgnoreCase("NULL")) {
+            return parsedConfig.get(MarkLogicSourceConfig.KEY_STRATEGY).toString().equalsIgnoreCase("UUID") ?
+                new UuidKeyGenerator() : new TimestampKeyGenerator(rowNumber);
```

Comment:
Nitpick - Maybe it's just me, but with this formatting, I find it a bit tricky to notice the TimestampKeyGenerator constructor call.

Reply:
What I do when using "?" is: 

```
return something that returns true or false ? 
    this thing :
    this other thing;
```

I find that usually reads pretty well.